java/execution/openapi/src/com/intellij/execution/ApplicationRunLineMarkerHider.java

```diff
@@ -0,0 +1,28 @@
+// Copyright 2000-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.intellij.execution;
+
+import com.intellij.openapi.extensions.ExtensionPointName;
+import com.intellij.psi.PsiElement;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * An extension point to validate whether the application run line marker should be displayed.
+ */
+public interface ApplicationRunLineMarkerHider {
+
+  ExtensionPointName<ApplicationRunLineMarkerHider> EP_NAME =
```

Comment:
Can be private

Reply:
I get a compiler error for making this private "Modifier 'private' not allowed here."